pyspeckit/spectrum/units.py

```diff
@@ -319,14 +319,17 @@ def __new__(self, xarr, unit='Hz', refX=None, redshift=None,
         subarr._unit = self.validate_unit(unit, bad_unit_response)
         subarr.refX = refX
 
+        if hasattr(refX, 'unit'):
+            if refX_unit != refX.unit:
+                raise ValueError("If refX is a Quantity then refX_unit must be None or the same as refX.unit")
+            refX_unit = refX.unit 
```

Comment:
Hey @keflavich, this allows the refX_unit to be the same with refX.unit in case a user for some reason passed both, if it is different though the ValueError is raised. This is a bit more forgiving, let me know what you think.